cruise-control-core/src/test/java/com/linkedin/cruisecontrol/monitor/sampling/aggregator/MetricSampleAggregatorTest.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.monitor.sampling.aggregator;
+
+import com.linkedin.cruisecontrol.CruiseControlUnitTestUtils;
+import com.linkedin.cruisecontrol.IntegerEntity;
+import com.linkedin.cruisecontrol.exception.NotEnoughValidWindowsException;
+import com.linkedin.cruisecontrol.metricdef.MetricDef;
+import com.linkedin.cruisecontrol.metricdef.MetricInfo;
+import com.linkedin.cruisecontrol.metricdef.AggregationFunction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+
+/**
+ * Unit test for {@link MetricSampleAggregator}
+ */
+public class MetricSampleAggregatorTest {
```

Comment:
Does this test checks the cases where it expects to see an exception in `MetricSampleAggregator`?

Reply:
We are testing it in another way by verifying the completeness. Do you have a specific test in mind?